include/hogl/detail/engine.hpp

```diff
@@ -67,6 +67,7 @@ class engine {
 		unsigned int tso_buffer_capacity;
 		unsigned int internal_ring_capacity; // obsolete
 		unsigned int features;
+		uint64_t     cpu_affinity_mask; // CPU affinity
```

Comment:
64bit is not enough for some machines these days :)
Should be cpu_set_t just like sched_setaffinity(). In glibc headers it can handle up to 1024 cores.